src/runtime/link-accounts-flow.js

```diff
@@ -194,41 +194,66 @@ export class LinkSaveFlow {
     /** @private @const {!Window} */
     this.win_ = deps.win();
 
+    /** @private @const {!./deps.DepsDef} */
+    this.deps_ = deps;
+
     /** @private @const {!web-activities/activity-ports.ActivityPorts} */
     this.activityPorts_ = deps.activities();
 
     /** @private @const {!../components/dialog-manager.DialogManager} */
     this.dialogManager_ = deps.dialogManager();
 
+    /** {!../api/subscriptions.SaveSubscriptionRequest} */
+    const defaultRequest = {token: 'test'};
+
     /** @private {!../api/subscriptions.SaveSubscriptionRequest} */
-    this.saveSubscriptionRequest_ = saveSubscriptionRequest;
+    this.saveSubscriptionRequest_ = saveSubscriptionRequest || defaultRequest;
 
     /** {!boolean} */
     this.completed_ = false;
 
-    /** @private @const {!ActivityIframeView} */
-    this.activityIframeView_ = new ActivityIframeView(
-        this.win_,
-        this.activityPorts_,
-        feUrl('/linksaveiframe'),
-        feArgs({
-          'publicationId': deps.pageConfig().getPublicationId(),
-          'token': this.saveSubscriptionRequest_['token'],
-        }),
-        /* shouldFadeBody */ false
-    );
+    /** @private {?ActivityIframeView} */
+    this.activityIframeView_ = null;
   }
 
   /**
    * Starts the save subscription
    * @return {!Promise}
    */
   start() {
-    this.activityIframeView_.acceptResult().then(() => {
-      this.completed_ = true;
-      // The flow is complete.
-      return this.dialogManager_.completeView(this.activityIframeView_);
+    this.activityIframeView_ = new ActivityIframeView(
+      this.win_,
+      this.activityPorts_,
+      feUrl('/linksaveiframe'),
+      feArgs({
+        'publicationId': this.deps_.pageConfig().getPublicationId(),
+        'token': this.saveSubscriptionRequest_['token'],
+        'isClosable': true,
+      }),
+      /* shouldFadeBody */ false
+    );
+    /** {!Promise<boolean>} */
+    const returnPromise = new Promise(resolve => {
+      const promise = this.activityIframeView_.port().then(port => {
+        return acceptPortResult(
+            port,
+            feOrigin(),
+            /* requireOriginVerified */ true,
+            /* requireSecureChannel */ true);
+      });
+      promise.then(result => {
```

Comment:
You don't need `new Promise` construct here. Just simple return this `promise` here. And move the `catch` and the final `then` close to the top level. It could be all nicely inlined on the `openView` promise below.

Reply:
Done.